mautic/mautic
# rename addFlash to addFlashMessage to prevent method collision

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [x]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

In Symfony 5, a helper method was added: `Symfony\Bundle\FrameworkBundle\Controller\AbstractController::addFlash()`, see [the source code](https://github.com/symfony/symfony/commit/30418cfaf3934a26698b7db220bdf909a4fbea2e#diff-af6672a1436a317ff73aea4f4bb77153b29876d8b9df1fe778425dba20bcd1c9R199).

This method collides with `Mautic\CoreBundle\Controller\CommonController::addFlash()` which extends `AbstractController::addFlash()`
There are 2 options: 
* changing the logic to make `CommonController::addFlash()` compatible with `AbstractController::addFlash()`
* renaming the method

As there is other business logic in Mautic (message is translated, possibility to add notification, ...), only the rename option is viable at this point.

This PR addresses this.
This PR contains BC breaking changes, so we need to document this




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. add a contact / company / ... (actually anything that shows a flash message after an action
3. verify the message is shown
